src/widget/wtrackmenu.cpp

```diff
@@ -185,6 +187,38 @@ void WTrackMenu::createMenus() {
                 });
     }
 
+    if (featureIsEnabled(Feature::FindOnWeb)) {
+        DEBUG_ASSERT(!m_pFindOnWebMenu);
+        m_pFindOnWebMenu = make_parented<WFindOnWebMenu>(this); //¿Is it okay to have parent menu?
+        connect(m_pFindOnWebMenu,
+                &QMenu::aboutToShow,
+                this,
+                [this] {
+                    m_pFindOnWebMenu->clear();
+                    const auto pTrack = getFirstTrackPointer();
+                    bool isSoundcloudEnabled = m_pConfig->getValue<bool>(
```

Comment:
Here we have the knowledge of the submenus in the WTrackMenu. 
I suggest to make the FindOnWebMenuFactory state-full and pass the config pointer during the constructor of the  WTrackMenu() 
The name of the config value shall also obtained form the FindOnDiscogsMenu()
The same should happen in the preferences. 
I think the Preferences GUI entries can also be created by the factory. 

In an ideal case, new Services can be added by cresting the cpp and header file and tell that the factory, every factory user shall not have own knowledge about the services.

Reply:
> Here we have the knowledge of the submenus in the WTrackMenu. I suggest to make the FindOnWebMenuFactory state-full and pass the config pointer during the constructor of the WTrackMenu().

Could you please explain more about state-full? If I don't know wrong, Config pointer is already passed to the WTrackMenu().

> The name of the config value shall also obtained form the FindOnDiscogsMenu() The same should happen in the preferences. I think the Preferences GUI entries can also be created by the factory.

Is it about the naming, or should I get the Config Value-user preferences in FindOnDiscogsMenu class? I'm a bit confused in this line too. I'm sorry.

I have tried to get these Config's value from the sub-menus, but it made the classes more complicated. I found myself adding this pointer to every service's class.

I have tried to get Config's value from the Factory class, It wasn't possible for me to use that Config Pointer, because all the methods were static in the Factory Class. But if it is a must I can try everything to do that.


> In an ideal case, new Services can be added by cresting the cpp and header file and tell that the factory, every factory user shall not have own knowledge about the services.

What I thought in the first place, we can have a main method in WFindOnWebMenu, and we can call this method in Services, constructors, so It will make so easy for us to add new services. If the service has different pagination, or query type, we can just override the main method. Should I do that? Or should I do something different?

But even right now, it is getting easier to add services.